### PR TITLE
Make Identity Server tests parallel and isolated

### PR DESCRIPTION
**Pull request by htdvisser**
#### Summary

This pull request makes a number of tests in the Identity Server execute in parallel, with isolated database schemas. There are still some leftovers, which I've listed in https://github.com/TheThingsNetwork/lorawan-stack/issues/5333.

#### Changes

- Lots of test refactoring; almost no changes in production code.

#### Testing

It was all about testing.

##### Regressions

nil

#### Notes for Reviewers

nil

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Compatibility: The changes are backwards compatible with existing API, storage, configuration and CLI, according to the compatibility commitments in `README.md` for the chosen target branch.
- [ ] Documentation: Relevant documentation is added or updated.
- [ ] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [x] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

**Review by johanstokking — APPROVED** [review]
Nice

File: pkg/identityserver/application_registry_test.go [hunk 1]
```diff
@@ -219,12 +229,6 @@ func TestApplicationsCRUD(t *testing.T) {
 			}
 		}
 
-		// Check that returned value is not nil
-		devEUIResponse, err := reg.IssueDevEUI(ctx, created.GetIds(), userCreds(adminUserIdx))
-		a.So(err, should.BeNil)
-		a.So(devEUIResponse, should.NotBeNil)
-		a.So(devEUIResponse.DevEui, should.NotBeZeroValue)
-
```
Comment (pgalic96) [inline pkg/identityserver/application_registry_test.go:230]: Where do we test this then?

**Review by pgalic96 — COMMENTED** [review]
Overall LGTM, I think you removed one test and I am not sure where the test should reside with this refactoring

File: pkg/identityserver/application_registry_test.go (hunk 1 above)
Comment (htdvisser) [reply pkg/identityserver/application_registry_test.go:230]: The logic itself is tested by the `storetest` package. The only additional logic in the IS implementation is the rights check. I've added it to [the list](https://github.com/TheThingsNetwork/lorawan-stack/issues/5333).

**Pull request merged**